subworkflows/local/polishing.nf

```diff
@@ -0,0 +1,88 @@
+
+workflow POLISHING {
+    take:
+    bam       // path: /path/to/bam
+    bam_bai   // path: /path/to/bai
+    fasta     // path: /path/to/fasta
+    fasta_fai // path: /path/to/fai
+    groups    // val: number of chunks
+    summary   // path: /path/to/longranger/summary/file
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Split genome into chunks
+    BED_CHUNKS (fasta_fai, groups)
+    ch_versions = ch_versions.mix(BED_CHUNKS.out.versions)
+    intervals_structured = BED_CHUNKS.out.coords.toList().transpose()
+    chunks_ch = Channel.from(bam)
+    intervals_freebayes = chunks_ch.combine(intervals_structured)
+     .map{ bam, bed -> [[id: bed.getSimpleName()], bam, bam_bai, [], [], bed] }
+
+    // In case the average coverage from Longranger is provided use it for defining 
+    // max coverage cut-off then scatter Freebayes over the genome chunks
+    if ( summary ) {
+        LONGRANGER_COVERAGE(summary)
+        ch_versions = ch_versions.mix(LONGRANGER_COVERAGE.out.versions)
+        FREEBAYES(intervals_freebayes, fasta, fasta_fai, [], [], [], LONGRANGER_COVERAGE.out.cov)
+    }
+    else {
+        FREEBAYES(intervals_freebayes, fasta, fasta_fai, [], [], [], "")
+    }
+    ch_versions = ch_versions.mix(FREEBAYES.out.versions)
+    BCFTOOLS_INDEX_FB(FREEBAYES.out.vcf)
+    ch_versions = ch_versions.mix(BCFTOOLS_INDEX_FB.out.versions)
+
+    // Filter and sort vcf for each genome chunk
+    FREEBAYES.out.vcf.map{ meta, vcf -> [meta.id.toString(), vcf]} 
+            .join(BCFTOOLS_INDEX_FB.out.tbi.map {meta, tbi -> [meta.id.toString(), tbi]})
+            .map{ id, vcf, tbi -> [[ id: id.toString()+'_view'], vcf, tbi ]}
+            .set{ input_view }
+    BCFTOOLS_VIEW(input_view, [], [], [])
+    ch_versions = ch_versions.mix(BCFTOOLS_VIEW.out.versions)
+    input_sort = BCFTOOLS_VIEW.out.vcf.map{ meta, vcf -> [ [id: meta.id.toString()+'_sorted'], vcf ]}
+    BCFTOOLS_SORT(input_sort)
+    ch_versions = ch_versions.mix(BCFTOOLS_SORT.out.versions)
+
+    // Merge vcf files into one
+    MERGE_FREEBAYES(BCFTOOLS_SORT.out.vcf.map{ meta, vcf -> [[id: 'merged'], vcf]}.groupTuple(), [])
+    ch_versions = ch_versions.mix(MERGE_FREEBAYES.out.versions)
+
+    // Normalize variants and index normalized vcf
+    MERGE_FREEBAYES.out.vcf.map{ meta, vcf -> [meta.id.toString(), vcf]}
+        .join(MERGE_FREEBAYES.out.tbi.map{ meta, tbi -> [meta.id.toString(), tbi]})
+        .map{ id, vcf, tbi -> [[ id: 'norm'], vcf, tbi ]}
+        .set{ input_norm }
+    BCFTOOLS_NORM(input_norm, fasta)
+//    BCFTOOLS_NORM(MERGE_FREEBAYES.out.vcf.map{meta, vcf -> [[id:'norm'], vcf]}, fasta)
+    ch_versions = ch_versions.mix(BCFTOOLS_NORM.out.versions)
+    BCFTOOLS_INDEX_NORM(BCFTOOLS_NORM.out.vcf)
+    ch_versions = ch_versions.mix(BCFTOOLS_INDEX_NORM.out.versions)
+
+    // Generate consensus fasta file    
+    ch_fasta = Channel.of([[id:'norm'], fasta])
```

Comment:
`id` is generally the organism ID, why are you changing to `norm` (a fixed value)? I can't see the advantage. This would change the `meta.id` for all downstream processes and affect the name of output files written back to `/lustre`

Reply:
Right, I see what you mean. I wonder where the appropriate id should be taken from here? Should it be specified somewhere in the higher level and passed over into the workflow as a var?

Reply:
There should be a way to pass the id somehow elegantly, but whenever I pass it with one of the vars as the channel like `[meta, fasta]` it breaks the rest of the code. When I then  try to do `fasta_ch.map{ meta, fasta -> FREEBAYES(..., fasta, ...)}` it breaks further on...